pages/api/email/[list].ts

```diff
@@ -24,52 +99,30 @@ export default async function handler(
       message: 'Email API: Invalid http method. | Only POST is accepted.',
     });
   }
-  const list = `${req.query.list}`;
-  const email = req.body.email;
 
-  const listId =
-    list === 'session'
-      ? process.env.CAMPAIGN_MONITOR_LIST_SESSION_ID
-      : process.env.CAMPAIGN_MONITOR_LIST_MARKET_RESEARCH_ID;
+  const list: EmailLists = `${req.query.list}` as EmailLists;
+  const email = req.body.email;
 
-  const body: any = {
-    EmailAddress: email,
-    ConsentToTrack: 'Unchanged',
-  };
+  let response;
 
-  if (list !== 'session') {
-    const roles = handleMultiValueSelect(req.body.roles, 'Roles') ?? [];
-    const tags = handleMultiValueSelect(req.body.tags, 'Tags') ?? [];
-
-    body['Name'] = req.body.name ? req.body.name : undefined;
-    body['CustomFields'] = [
-      req.body.country && {
-        Key: 'Country',
-        Value: req.body.country,
-      },
-      ...roles,
-      ...tags,
-    ];
-  }
+  if (list === 'market-research') {
```

Comment:
Are we fine with this being user controlled? Can anyone elect to join the `market-research` list?

Reply:
No they cannot. The component that hits the api with /api/email/market-research is only rendered when a specific shortcode is used in contentful.

Reply:
but theres nothing stopping people doing `curl -X POST .../api/email/market-research ...`

If we dont mind then thats fine.

Reply:
discussed in person